pkg/stacks/truncate/truncate_test.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package truncate provides functions for truncating Kubernetes values in a
+// predictable way offering safe values usable in deterministic field searches.
+package truncate
+
+import (
+	"strings"
+	"testing"
+)
+
+var (
+	str63  = strings.Repeat("1234567890", 6) + "123"
+	str253 = strings.Repeat("1234567890", 25) + "123"
+)
+
+func Test_truncate(t *testing.T) {
+	type args struct {
+		str          string
+		length       int
+		suffixLength int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name:    "StrTooShort",
+			args:    args{str: "a", length: 0, suffixLength: 0},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "SuffixExceedsStr",
+			args:    args{str: "a", length: 0, suffixLength: 2},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "SuffixExceedsSum",
+			args:    args{str: strings.Repeat("12345678890", 21), length: 200, suffixLength: 100},
```

Comment:
This is a dumb question, but did we consider just padding out to the desired suffix length? : )

Reply:
I didn't consider that. For our purposes the truncation suffix is locked in.  I added this test and the error response it invokes for logical completeness.